fhir-search/src/main/java/com/ibm/fhir/search/parameters/ParametersUtil.java

```diff
@@ -79,185 +64,95 @@
-        return populateSearchParameterMapFromStreamByFormat(stream, Format.XML);
-    }
-
-    /*
-     * Loads SearchParameters using the specified InputStream and returns a Map containing them.
-     * @param stream the InputStream from which to load the SearchParameters
-     * @param Format
-     * @return
+     * @param bundle a Bundle of type Collection with entries of type SearchParameter
+     * @return a Map of ParameterMaps, keyed by resourceType
+     * @throws ClassCastException if the Bundle contains entries of any type other than SearchParameter
      */
-    private static Map<String, Map<String, SearchParameter>> populateSearchParameterMapFromStreamByFormat(InputStream stream, Format format) {
-        // Format is never null, as this method is private and hidden.
-        // In order to maintain this contract, and avoid the check, keep private.
-
-        Map<String, Map<String, SearchParameter>> searchParameterMap = new HashMap<>();
-
-        // The code block is isolating the effects of exceptions by capturing the exceptions here
-        // and returning an empty searchParameterMap, and continuing operation.
-        // The failure is logged out.
-        try {
-            // The code is agnostic to format.
-            Bundle bundle = FHIRParser.parser(format).parse(stream);
-
-            FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
-            EvaluationContext evaluationContext = new EvaluationContext(bundle);
-            Collection<FHIRPathNode> result = evaluator.evaluate(evaluationContext, FHIR_PATH_BUNDLE_ENTRY);
-
-            List<SearchParameter> paramList = result.stream()
-                    .map(node -> node.asResourceNode().resource().as(SearchParameter.class))
-                    .collect(Collectors.toList());
-            for (SearchParameter parameter : paramList) {
+    public static Map<String, ParametersMap> buildSearchParametersMapFromBundle(Bundle bundle) {
+        Map<String, ParametersMap> typeToParamMap = new HashMap<>();
+        
+        for (Bundle.Entry entry : bundle.getEntry()) {
+            SearchParameter parameter = entry.getResource().as(SearchParameter.class);
+            
+            // Conditional Logging intentionally avoids forming of the String.
+            if (log.isLoggable(Level.FINE)) {
+                log.fine(String.format(LOG_PARAMETERS, parameter.getCode().getValue()));
+            }
 
-                // Conditional Logging intentionally avoids forming of the String.
+            if (parameter.getExpression() == null || !parameter.getExpression().hasValue()) {
                 if (log.isLoggable(Level.FINE)) {
-                    log.fine(String.format(LOG_PARAMETERS, parameter.getCode().getValue()));
+                    log.fine(String.format(MISSING_EXPRESSION, parameter.getCode().getValue()));
                 }
+                continue;
+            }
```

Comment:
I suggest explicitly adding an else block in the loop.  continues can be lost in a complicated code block.

Reply:
I generally prefer the early exit (or continue in this case) for the exception cases...I don't like long else blocks which lead to extra level of indentation and I forget what was the if/else conditions above.  Just a difference in style preference I suppose, but in this case I went with the suggestion.